storage/tianmu/loader/load_parser.cpp

```diff
@@ -175,15 +175,15 @@ bool LoadParser::MakeValue(uint att, ValueCache &buffer) {
   // deal with auto increment
   auto &attr(attrs_[att]);
   if (core::ATI::IsIntegerType(attrs_[att]->TypeName()) && attr->GetIfAutoInc()) {
-    uint64_t *buf = reinterpret_cast<uint64_t *>(buffer.Prepare(sizeof(uint64_t)));
-    uint64_t value = *(uint64_t *)buf;
+    int64_t *buf = reinterpret_cast<int64_t *>(buffer.Prepare(sizeof(uint64_t)));
 
-    if (value == 0)  // Value of auto inc column was not assigned by user
+    if (*buf == 0)  // Value of auto inc column was not assigned by user
       *buf = attr->AutoIncNext();
 
-    if (value > attr->GetAutoInc())
-      attr->SetAutoInc(value);
-
+    if (static_cast<uint64_t>(*buf) > attr->GetAutoInc()) {
+      if (*buf > 0 || ((attr->TypeName() == common::ColumnType::BIGINT) && attr->GetIfUnsigned()))
```

Comment:
Whether `*buf >0` statement can be move back to `if (*buf ==0)` statement to test the condition as soon as possible, and therefore, at this line we can only do typename and unsigned flag  check.

Reply:
`*buf ==0` will acquire the auto_increment value, And ` *buf > 0` is in order to assign the next start value to the auto_increment. In addition,` *buf < 0` will not change auto_increment value, which is consistent with innodb，except unsigned bigint type which is greater than `INT64_MAX`.
When col_value  is greater than `INT64_MAX`, `*buf `will is negative.

Reply:
For example:
`insert into auto_table values(0, "nnn");  # *buf == 0`
`insert into auto_value values(1,"sss"); # * buf > 0`
`insert into auto_value values(9223372036854775809, "dddd"); # 9223372036854775809> INT64_MAX, *buf <0`